cpp/src/fil/fil.cu

```diff
@@ -408,12 +417,16 @@ void check_params(const forest_params_t* params, bool dense) {
              "softmax does not make sense for leaf_algo == FLOAT_UNARY_BINARY");
       break;
     case leaf_algo_t::GROVE_PER_CLASS:
+      ASSERT(params->threads_per_tree == 1,
```

Comment:
It's worth adding to the pull request description that it doesn't currently work with multi-class.